src/aks-preview/azext_aks_preview/custom.py

```diff
@@ -1416,6 +1426,10 @@ def aks_update(cmd,     # pylint: disable=too-many-statements,too-many-branches,
     if disable_ahub:
         instance.windows_profile.license_type = 'None'
 
+    if auto_upgrade_channel is not None:
+        if instance.auto_upgrade_profile is None:
+            instance.auto_upgrade_profile = ManagedClusterAutoUpgradeProfile()
```

Comment:
ha None in python is a little confusing with your None upgrade channel :).
Nit I find it a little more fool proof to always fill in structs like auto_upgrade_profile if they're blank. That way if there's another argumetn later it doesn't have to do the same. 

        if instance.auto_upgrade_profile is None:
            instance.auto_upgrade_profile = ManagedClusterAutoUpgradeProfile()

        if auto_upgrade_channel is not None:
              instance.auto_upgrade_profile.upgrade_channel = auto_upgrade_channel

Reply:
yea, it's a bit confusing, :) updated the logic as suggested.